hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -285,7 +285,8 @@ public DeltaSync(HoodieDeltaStreamer.Config cfg, SparkSession sparkSession, Sche
     // Register User Provided schema first
     registerAvroSchemas(schemaProvider);
 
-    this.transformer = UtilHelpers.createTransformer(Option.ofNullable(cfg.transformerClassNames));
+    this.transformer = UtilHelpers.createTransformer(Option.ofNullable(cfg.transformerClassNames),
```

Comment:
I want to call out by fetching the source schema here, we are doing a one-time schema validation. I'm trying to think through the impact of this when columns are added/dropped. This may be fine but wanted to make sure this was an intentional decision.

Reply:
We'll never factor these new fields into the validation. Is that ok?